cli/connhelper/connhelper.go

```diff
@@ -47,7 +47,7 @@ func getConnectionHelper(daemonURL string, sshFlags []string) (*ConnectionHelper
 		}
 		return &ConnectionHelper{
 			Dialer: func(ctx context.Context, network, addr string) (net.Conn, error) {
-				return commandconn.New(ctx, "ssh", append(sshFlags, sp.Args("docker", "system", "dial-stdio")...)...)
+				return commandconn.New(ctx, "ssh", append([]string{"-o ConnectTimeout=30"}, append(sshFlags, sp.Args("docker", "system", "dial-stdio")...)...)...)
```

Comment:
This reads rather awkward; also looking;

- `getConnectionHelper()` is called both by `GetConnectionHelper()` and `GetConnectionHelperWithSSHOpts()`
- `-o ConnectTimeout=30` is an `sshFlag` ?

Wondering; should this option be set _both_ for `GetConnectionHelper` and for `GetConnectionHelperWithSSHOpts` (which we currently don't use ourselves, but others may be using it)

Some options;

If only for `GetConnectionHelper`, then we could pass it as option when calling `getConnectionHelper()`

```go
func GetConnectionHelper(daemonURL string) (*ConnectionHelper, error) {
	return getConnectionHelper(daemonURL, []string{"-o ConnectTimeout=30"})
}
```

If it _must_ be set for both, then we could add it here to `sshFlags`;

```go
		sshFlags = append([]string{"-o ConnectTimeout=30"}, sshFlags...)
		return &ConnectionHelper{
			Dialer: func(ctx context.Context, network, addr string) (net.Conn, error) {
				return commandconn.New(ctx, "ssh", append(sshFlags, sp.Args("docker", "system", "dial-stdio")...)...)
			},
			Host: "http://docker.example.com",
		}, nil
```


⚠️ ☝️ what happens if someone uses `GetConnectionHelperWithSSHOpts()`, and sets `-o ConnectTimeout=30` as custom option?

```go
GetConnectionHelperWithSSHOpts("ssh://foo", []string{"-o ConnectTimeout=60"})
```

In that case (with this PR), `ssh` will be called with _both_ `-o ConnectTimeout=30` and `-o ConnectTimeout=60`; does that produce an error?

Reply:
> In that case (with this PR), `ssh` will be called with both `-o ConnectTimeout=30` and `-o ConnectTimeout=60`; does that produce an error?


Looks like it doesn't complain (not sure which timeout is used though? first? last?)

```bash
ssh -o ConnectTimeout=30 -o ConnectTimeout=60 swarm-test-01
...
```

💁 If we decide to apply it to both, it may be worth mentioning the default timeout in the function's descriptions.

Reply:
```go
sshFlags = append([]string{"-o ConnectTimeout=30"}, sshFlags...)
		return &ConnectionHelper{
			Dialer: func(ctx context.Context, network, addr string) (net.Conn, error) {
				return commandconn.New(ctx, "ssh", append(sshFlags, sp.Args("docker", "system", "dial-stdio")...)...)
			},
			Host: "http://docker.example.com",
		}, nil
```
Fun fact, if you do it this way and then a connection gets dialed 3 times you end with
```console
$ ssh -o ConnectTimeout=30 -o ConnectTimeout=30 -o ConnectTimeout=30 ...
```
(ask me how I know)

In reality though, I iterated through a few different things while making this change (a `getSSHTimeoutArgs` func that also adds an env var for users to be able to disable/adjust it, checking if the command passed in already includes a `ConnectTimeout` option and only adding it if it wasn't there already, etc), but ended up just using this since it didn't seem like this was getting called by anyone else.

I should've considered that other people may be calling it though! I'll take a whack at it again, maybe go with the simple implementation and only add it if it wasn't already specified.